Tools/clinic/clinic.py

```diff
@@ -2412,26 +2412,22 @@ def _module_and_class(
         this function is only ever used to find the parent of where
         a new class/module should go.
         """
-        parent: Clinic | Module | Class
-        child: Module | Class | None
-        module: Clinic | Module
+        parent: Clinic | Module | Class = self
+        module: Clinic | Module = self
         cls: Class | None = None
-        so_far: list[str] = []
 
-        parent = module = self
-
-        for field in fields:
-            so_far.append(field)
+        for idx, field in enumerate(fields):
             if not isinstance(parent, Class):
-                child = parent.modules.get(field)
-                if child:
-                    parent = module = child
+                if field in parent.modules:
+                    module = parent.modules[field]
+                    parent = module
```

Comment:
I actually find putting both assignments on one line more readable -- conceptually, you're assigning both variables to the same thing here (`parent.modules[field]`), rather than assigning `module` to `parent.modules[field]`, and then assigning `module` to whatever `parent` is:

```suggestion
                    parent = module = parent.modules[field]
```